Cython/Compiler/Main.py

```diff
@@ -92,7 +94,7 @@ def __init__(self, include_directories, compiler_directives, cpp=False,
     @classmethod
     def from_options(cls, options):
         return cls(options.include_path, options.compiler_directives,
-                   options.cplus, options.language_level, options=options)
+                   options.cplus, options.language_level, options.shared_utility_qualified_name, options=options)
```

Comment:
Seeing this, it really makes no sense. The `Context` knows its `.options` and we're only duplicating the option into the `Context` here.  Just go through the options then. That will mean that several places that now pass the `Context` down into a function can instead pass only `context.options`. Again, less dependencies.